src/lib/steps/setupPages.js

```diff
@@ -8,13 +9,17 @@ const withoutPropsSetup = require('../pages/withoutProps/setup')
 
 // Set up all our NextJS pages according to the recipes defined in pages/
 const setupPages = async ({ functionsPath, publishPath }) => {
-  await apiSetup(functionsPath)
-  await getInitialPropsSetup(functionsPath)
-  await getServerSidePropsSetup(functionsPath)
-  await getStaticPropsSetup({ functionsPath, publishPath })
-  await getSPFallbackSetup(functionsPath)
-  await getSPRevalidateSetup(functionsPath)
-  await withoutPropsSetup(publishPath)
+  const jobs = [
+    ...(await apiSetup(functionsPath)),
+    ...(await getInitialPropsSetup(functionsPath)),
+    ...(await getServerSidePropsSetup(functionsPath)),
+    ...(await getStaticPropsSetup({ functionsPath, publishPath })),
+    ...(await getSPFallbackSetup(functionsPath)),
+    ...(await getSPRevalidateSetup(functionsPath)),
+    ...(await withoutPropsSetup(publishPath)),
+  ]
+
+  return runJobsQueue(jobs)
```

Comment:
Because `runJobsQueue` returns a Promise, returning it means we can `await` the return value outside. It's similar to just `await`ing `runJobsQueue()` before returning, but saves us creating another Promise.